packages/blade/src/components/StepGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,219 @@
+        title="Introduction"
+        timestamp="Thu, 11th Oct23 | 12:00pm"
+        isSelected={selectedIndex === 0}
+        // `onClick` prop turns item into Interactive item
+        onClick={() => {
+          setSelectedIndex(0);
+        }}
+      />
+      <StepItem
+        leading={<StepItemIcon icon={ClockIcon} color="primary" />}
+        title="Compliance Details"
+        description="Provide documentation of reports"
+        isSelected={selectedIndex === 1}
+        onClick={() => {
+          setSelectedIndex(1);
+        }}
+      />
+    </StepGroup>
+  );
+};
+```
+
+### Nested StepGroup
+
+<img width="200px" src="./2024-04-12-15-22-54.png" alt="" />
+
+```jsx
+<StepGroup>
+  <StepItem
+    leading={<StepItemIcon icon={CheckIcon} color="positive" />}
+    title="Disputes Raised"
+    timestamp="Thu, 11th Oct23 | 12:00pm"
+    stepProgress="full"
+    trailing={<Badge>Label</Badge>}
+  />
+  {/* Nested StepGroup */}
+  <StepGroup>
+    <StepItem
+      leading={<StepItemIcon icon={CheckIcon} color="primary" />}
+      title="Needs Response"
+      timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+      description="Description"
+      stepProgress="start"
+    />
+    <StepItem
+      leading={<StepItemIcon icon={CheckIcon} color="neutral" />}
+      title="Needs Response"
+      timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+      description="Description"
+    />
+  </StepGroup>
+  <StepItem
+    leading={<StepItemIcon icon={CheckIcon} color="neutral" />}
+    title="Decision from Bank"
+    trailing={<Badge color="neutral">Pending</Badge>}
+  />
+</StepGroup>
+```
+
+### Horizontal Orientation
+
+<img width="400px" src="./image.png" />
+
+```jsx
+<StepGroup orientation="horizontal">{/* StepItem components */}</StepGroup>
+```
+
+## Accessibility
+
+- Make all steps visitable with `TAB` in interactive StepGroup
+  - Render StepItem as `a` tag when `href` is used
+  - Render StepItem as `button` tag when `onClick` is used without `href`
+- StepGroup does not need any separate role to be defined or special tag to be used.
+
+## References
+
+- [ProgressSteps - Baseweb](https://baseweb.design/components/progress-steps/)
+- [React Stepper - MUI](https://mui.com/material-ui/react-stepper/)
+- [Stepper - ZenDesk](https://garden.zendesk.com/components/stepper)
+
+All follow a similar compound API.
+
+## Open Questions
+
+- ### Should we add `isInteractive` prop?
+
+  In the proposed API, I have proposed that we can turn item into interactive or static based on whether it has `onClick` or `href` or nothing. Is it intuitive enough? or should we add more explict prop called `isInteractive` like we have in design
+
+- ### Alternative to `leading`
+
+  Currently I have proposed `leading` prop where we can add Icon or indicator. Although in horizontal orientation, its not exactly "leading". It comes on top. Its also not very equivalent to leading we have in other components.
```

Comment:
what happens to trailing on horizontal variant?

Reply:
There is no trailing in horizontal